### PR TITLE
SYN-7610, SYN-7457: Multiple small model updates

### PR DESCRIPTION
**Pull request by MichaelSquires**
feat: Add ``entity:name`` form.
feat: Update ``ou:position:title`` to be an ``ou:jobtitle`` type instead of ``str``.
feat: Add model migration to update ``ou:conference:name`` from ``str`` to ``entity:name``.
feat: Add model migration to update ``ou:position:title`` from ``str`` to ``ou:jobtitle``.

SYN-7610: create entity:name and migrate conference names
SYN-7457: ou:position:title migrated to ou:jobtitle

### CODE REVIEW

File: synapse/models/entity.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import synapse.lib.module as s_module
+
+class EntityModule(s_module.CoreModule):
+
+    def getModelDefs(self):
+        return (('entity', {
+
+            'types': (
+                ('entity:name', ('str', {'onespace': True, 'lower': True, 'strip': True}), {
```
Comment (MichaelSquires) [inline synapse/models/entity.py:9]: @invisig0th I added the ``strip`` opt here too. I can easily back it out if you want.

**Comment by MichaelSquires** [comment]
Regression repo PR: https://github.com/vertexproject/synapse-regression/pull/21

File: synapse/lib/modelrev.py [hunk 2]
```diff
@@ -757,6 +758,10 @@ async def revModel_0_2_24(self, layers):
         for form, props in formprops.items():
             await self._normVelocityProps(layers, form, props)
 
+    async def revModel_0_2_25(self, layers):
+        await self._normPropValu(layers, 'ou:conference:name')
```
Comment (invisig0th) [inline synapse/lib/modelrev.py:762]: I believe these also need calls to generate the forms?

File: synapse/models/entity.py (hunk 1 above)
Comment (invisig0th) [reply synapse/models/entity.py:9]: onespace is a super-set of strip, so it's redundant. Remove plz :)

File: synapse/models/orgs.py [hunk 3]
```diff
@@ -731,8 +731,7 @@ def getModelDefs(self):
                     ('contact', ('ps:contact', {}), {
                         'doc': 'The contact info for the person who holds the position.',
                     }),
-                    # TODO migrate to ou:jobtitle
-                    ('title', ('str', {'lower': True, 'onespace': True}), {
+                    ('title', ('ou:jobtitle', {'lower': True, 'onespace': True}), {
```
Comment (invisig0th) [inline synapse/models/orgs.py:734]: `ou:jobtitle` is already lower/onespace so it should not be defined here.

File: synapse/tests/test_lib_modelrev.py [hunk 4]
```diff
@@ -473,3 +473,26 @@ async def test_modelrev_0_2_24(self):
...
+            ))
+
+            nodes = await core.nodes('ou:position')
+            self.len(3, nodes)
+            titles = [n.get('title') for n in nodes]
+            self.sorteq(titles, (
+                'president of the united states',
+                'vice president of the united states',
+                'speaker of the house, united states congress',
+            ))
```
Comment (invisig0th) [inline synapse/tests/test_lib_modelrev.py:498]: Need tests that confirm the `ou:jobtitle` and `entity:name` forms have also been created and are pivotable :+1:

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/modelrev.py [hunk 5]
```diff
@@ -757,6 +758,11 @@ async def revModel_0_2_24(self, layers):
         for form, props in formprops.items():
             await self._normVelocityProps(layers, form, props)
 
+    async def revModel_0_2_25(self, layers):
+        await self._propToForm(layers, 'ou:position:title', 'ou:jobtitle')
+        await self._propToForm(layers, 'ou:conference:name', 'entity:name')
+        await self._propArrayToForm(layers, 'ou:conference:names', 'entity:name')
```
Comment (Cisphyx) [inline synapse/lib/modelrev.py:764]: You'll need to norm these two first since they're now `onespace: True`

**Review by mikemoritz — DISMISSED** [review]
LGTM - just a reminder to remove the synapse-regression branch

**Review by invisig0th — APPROVED** [review]

**Review by invisig0th — CHANGES_REQUESTED** [review]
looks good. remove the branch build and we'll merge today!

**Review by invisig0th — APPROVED** [review]

**Pull request merged**